packages/camera/camera_web/example/integration_test/camera_web_test.dart

```diff
@@ -535,29 +535,91 @@ void main() {
 
           expect(
             (CameraPlatform.instance as CameraPlugin).cameras[cameraId],
-            isA<Camera>()
```

Comment:
Another way of doing this would be:

```
final Camera? camera = (CameraPlatform.instance as CameraPlugin).cameras[cameraId];

expect (camera, isNotNull);
expect (camera!.textureId, cameraId);
expect (camera.options.audio.enabled, isTrue);
```

(I don't know if there's getters for `options.audio`, but you get the idea.)

I'd only use `having` for things that need to be "computed" to be verified (which normally there aren't a lot of!)

Reply:
Refactored as:
```dart
final Camera? camera =
    (CameraPlatform.instance as CameraPlugin).cameras[cameraId];

expect(camera, isA<Camera>());
expect(camera!.textureId, cameraId);
expect(camera.options.audio.enabled, isTrue);
expect(camera.options.video.facingMode,
    equals(FacingModeConstraint(CameraType.user)));
expect(camera.options.video.width!.ideal,
    ultraHighResolutionSize.width.toInt());
expect(camera.options.video.height!.ideal,
    ultraHighResolutionSize.height.toInt());
expect(camera.options.video.deviceId, cameraMetadata.deviceId);
```